### DISCUSSION TITLE
Using the new SQLAlchemy implementation requires an async driver thats not provided when following the docs.

### DISCUSSION THREAD

**Discussion by Hazedd** [fastapi-users/fastapi-users#841] [Q&A] [answered]
I'm migrating to the new 9.x version of fastapi-users. The previous adapter using encode/databases is now deprecated and there for im trying to implementing the asyncio driver.  I have followed the documentation (https://fastapi-users.github.io/fastapi-users/configuration/databases/sqlalchemy/) 

on:

``` 
engine = create_async_engine(DATABASE_URL, connect_args={"check_same_thread": False})
```

i get the following error:
```
  File "....\backend\venv\lib\site-packages\sqlalchemy\ext\asyncio\engine.py", line 621, in __init__
    raise exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: The asyncio extension requires an async driver to be used. The loaded 'pysqlite' is not async.
```

I'm not sure, but i think it's a dependency issues, i was wondering if someone has had this issue aswel.

## Configuration
 - Python version : 3.10.1
 - FastAPI version : 0.70.1
 - FastAPI Users version: 9.2.0

**Comment by frankie567** [comment, marked as answer]
You should install and use an asyncio driver for your DBMS. It's [mentioned in the docs](https://fastapi-users.github.io/fastapi-users/configuration/databases/sqlalchemy/#asynchronous-driver).

For example, for SQLite:

```
pip install aiosqlite
```

```py
DATABASE_URL = "sqlite+aiosqlite:///./test.db"  # Notice the "+aiosqlite" to set the driver
```

**Reply by Hazedd** [reply]
thx! looked over the minor DATABASE_URL change.

**Comment by Hazedd** [comment]
@frankie567 i got the basic setup sorted register and login works. think the docs need a bit of work, still importing and using database in app.py.

Before implementing it in my angular project i made a fastapi-user full example repository so i know what i need to change. I all so including the oayth2 flow on google. i'm all most sure i followed the docs (could be wrong see above 🤔) and i facing a error on creating a user. 
```
sqlalchemy.exc.InvalidRequestError: When initializing mapper mapped class UserTable->user, expression 'OAuthAccount' failed to locate a name ('OAuthAccount'). If this is a class name, consider adding this relationship() to the <class 'app.db.UserTable'> class after both dependent classes have been defined.
```

i was wondering if you can take a look, maybe its worth to finish it as an example repository .

https://github.com/Hazedd/fastapi-users-example

edit:
all sorted now bit to fast copying stuff from the docs... will try to upgrade the docs in the next few dayz.
